docs/adr/0024-upgrade-flutter-3.24.md

```diff
@@ -0,0 +1,25 @@
+# 23. Flutter 3.24.0 Upgrade
+
+**Date:** 2024-08-26
+
+## Status
+
+**Accepted**
+
+## Context
+
+To enable features like text pasting without requiring user permission on iOS, we need to upgrade to Flutter 3.24.0.
+
+## Decision
+
+During the upgrade to Flutter 3.24.0, some dependencies must also be updated to ensure successful release builds. Specifically, the `targetSdkVersion` or `compileSdkVersion` needs to be updated to at least 31. Below is the list of packages that require a minimum `compileSdkVersion` of 31 or higher for successful compilation:
+
+- **`callkeep`**: Upgrade `compileSdkVersion` from 28 to 31. [See PR #190](https://github.com/flutter-webrtc/callkeep/pull/190)
+- **`receive_sharing_intent`**: Switch to the version maintained by Linagora, which has `compileSdkVersion` set to 33. [See Linagora Repository](https://github.com/linagora/receive_sharing_intent)
+- **`flutter_contacts`**: Upgrade `compileSdkVersion` from 31 to 33. [See PR #11](https://github.com/linagora/flutter_contacts/pull/11)
```

Comment:
Currently, we're using android 14 (compileSdkVersion is 34) I think should update also

Reply:
We don't have to, when the author update it, we will follow

<img width="720" alt="image" src="https://github.com/user-attachments/assets/6f112688-cfe6-4a61-948a-866b61210fdb">

Reply:
Pls deep thoughts!

Source: https://developer.android.com/google/play/requirements/target-sdk

Reply:
IMO, please update to 34